### PR TITLE
initial support for filter rules

### PR DESCRIPTION
**Pull request by toddjordan**
Addresses https://github.com/toddjordan/ember-cli-dynamic-forms/issues/1

### CODE REVIEW

**Comment by toddjordan** [comment]
@jeremywrowe do you mind reviewing this?  I refactored the component a bit to separate main functions out of the lifecycle hooks and make things a bit more clear.

**Comment by jeremywrowe** [comment]
Will do

File: addon/components/dynamic-form.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ const TYPE_MAP = {
   }
 };
 
+
```
Comment (jeremywrowe) [inline addon/components/dynamic-form.js]: Why the added spaces here?

File: tests/dummy/app/routes/index.js [hunk 2]
```diff
@@ -14,12 +16,22 @@ const basicObject = {
...
 };
 
 const basicString = JSON.stringify(basicObject);
 
+const basicFilterRule = _.clone(basicObject, true);
+
+basicFilterRule["options"] = {
+  "fields": {
+    "name": {
+      "filterrules": [ "admin-only" ]
```
Comment (jeremywrowe) [inline tests/dummy/app/routes/index.js]: Should this be `filterRules` to match js naming conventions?

**Comment by jeremywrowe** [comment]
Looks good, just some small tweaks / questions.

File: tests/dummy/app/routes/index.js (hunk 2 above)
Comment (toddjordan) [inline tests/dummy/app/routes/index.js]: I think to follow conventions we should rename to filter-rules instead. its both a directory and a json key, both of which we dasherize.

**Pull request merged**